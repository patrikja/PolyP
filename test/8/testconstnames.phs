data IC a = IC Int Char String deriving Show

main = print (inn x :: IC ())

polytypic x :: f a b
  = case f of
      g * h        -> (x, x) -- :: (g a b, h a b) 
      Const Int    -> 1
      Const Char   -> 'a'
      Const String -> "hej"
      
