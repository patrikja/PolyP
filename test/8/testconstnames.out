IC 1 'a' "hej"
